conan/tools/gnu/makedeps.py

```diff
@@ -0,0 +1,216 @@
+                    return False
+                line_buffer.append(self._var_assign_single(name, value))
+            else:
+                assert type(item) is list
+                ritem = _rootify(root, root_id, item) if root else item
+                if prefix_var:
+                    ritem = [self._var_ref(prefix_var) + i for i in ritem]
+                if len(ritem) == 0:
+                    return False
+                elif len(ritem) == 1:
+                    line_buffer.append(self._var_assign_single(name, ritem[0]))
+                else:
+                    line_buffer.append(self._var_assign_multi(name, ritem))
+            if blank_after:
+                line_buffer.append("")
+            return True
+
+        def _output_cpp_info_vars(var_id, cpp_info, root=None, root_id=None, add_to_aggregates=True, dep=None):
+
+            def _var(name, item, prefix_var=None):
+                v = f"CONAN_{name}_{var_id}"
+
+                if _output_var(v, item, root, root_id, prefix_var) and add_to_aggregates:
+                    var_aggregates.setdefault(name, []).append(self._var_ref(v))
+
+            prefix_path = root.replace("\\", "/")
+
+            # INFO: sysroot is a list with an empty string as default value
+            sysroot = cpp_info.sysroot if isinstance(cpp_info.sysroot, list) else [cpp_info.sysroot]
+            if sysroot and sysroot[0]:
+                sysroot = _get_formatted_dirs(sysroot, prefix_path, _makeify(dep.ref.name))
+                _var("SYSROOT", sysroot)
+
+            for var in ['INCLUDE_DIRS', 'LIB_DIRS', 'BIN_DIRS', 'SRC_DIRS', 'BUILD_DIRS', 'RES_DIRS',
+                        'FRAMEWORK_DIRS']:
+                cppinfo_value = getattr(cpp_info, var.replace('_', '').lower())
+                dirs = _get_formatted_dirs(cppinfo_value, prefix_path, _makeify(dep.ref.name))
+                _var(var, dirs, f"CONAN_{var}_FLAG")
+
+            # TODO: Collect from components if possible
+            common_variables = {
+                "OBJECTS": None,
+                "LIBS": "CONAN_LIB_FLAG",
+                "SYSTEM_LIBS": "CONAN_SYSTEM_LIB_FLAG",
+                "DEFINES": "CONAN_DEFINE_FLAG",
+                "CFLAGS": None,
+                "CXXFLAGS": None,
+                "SHAREDLINKFLAGS": None,
+                "EXELINKFLAGS": None,
+                "FRAMEWORKS": None,
+                "REQUIRES": None,
+            }
+
+            for var, prefix_var in common_variables.items():
+                cppinfo_value = getattr(cpp_info, var.lower())
+                # Use component cpp_info info when does not provide any value
+                if not cppinfo_value and hasattr(cpp_info, "components"):
+                    cppinfo_value = [self._var_ref(f"CONAN_{var}_{_makeify(var_id)}_{_makeify(name)}") for name, obj in cpp_info.components.items() if getattr(obj, var.lower())]
+                    # avoid repeating same prefix twice
+                    prefix_var = None
```

Comment:
To reiterate, I'm leery of this idea of synthesizing values for unset top-level package variables from the components. Do other generators do this?

While it poses no immediate problem for my usage (since I would ignore those variables anyway), I can see this causing headaches in maintaining packages with components. For example, a `foo` package maintainer starts out not setting `cpp_info.libs`, so we get the synthesized `CONAN_LIBS_FOO` aggregate value. People use the synthesized `CONAN_LIBS_FOO`, since it seems like the obvious thing to do. Later on, the maintainer decides to start setting `.libs` to some default library for the package (something that makes sense in context vis-a-vis the components). Now, people who were depending on the synthesized value get broken builds, as they may need to link in component libraries that are no longer listed in `CONAN_LIBS_FOO`.

And the maintainer is pressured to walk back the change. This discourages them from using a variable that they had never set before, because by not setting it, they were "opting in" to using this synthesized value.

IMHO, if it makes sense for `CONAN_LIBS_FOO` (or some other top-level package variable) to list the aggregated component values, then that should be set explicitly in the conanfile. Having the generator try to be "helpful" in this way is liable to cause problems down the road.